### PR TITLE
feat(Select): add creatable and new features

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Adds the creatable and new use cases to select via `isCreatable` and `onCreateOption` properties. Specifically applies to typeahead variants.

**Refer to issue**: #2665

### CODE REVIEW

**Comment by kmcfaul** [comment]
@mcarrano 
When a new item is entered, the addition occurs on the user-side. In the example, I just appended the option to the list. Select does not sort options.

Both new and creatable should work for multi-type ahead select as well. I will add the checkbox toggles to that example!

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectDemo.tsx [hunk 1]
```diff
@@ -277,6 +294,12 @@ export class SelectDemo extends Component<SelectDemoState> {
     }
   }
 
+  typeaheadCreateNew = (newValue: string) => {
```
Comment (dlabaj) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectDemo.tsx]: Should also allow SelectOptionObject.

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectDemo.tsx [hunk 2]
```diff
@@ -47,7 +50,15 @@ export class SelectDemo extends Component<SelectDemoState> {
...
     customTypeaheadMultiIsExpanded: false,
     customTypeaheadMultiSelected: [],
-    direction: SelectDirection.down
+    direction: SelectDirection.down,
+    typeaheadOptions: [
+      { value: 'Alabama', disabled: false},
+      { value: 'Florida', disabled: false },
+      { value: 'New Jersey', disabled: false }
+    ],
+    typeaheadCreateNew: undefined,
```
Comment (dlabaj) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectDemo.tsx]: Why does this need to be set to undefined?

Comment (kmcfaul) [reply packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/SelectDemo/SelectDemo.tsx]: It doesn't, I will remove

**Comment by jschuler** [comment]
I'm getting the browser autofill popup, should be turned off?
Screen Shot 2019-09-05 at 4 04 14 PM

**Comment by christiemolloy** [comment]
How is `isCreatable` different from `onCreateOption`?

When I have onCreateOption selected I can't create an option:
Screen Shot 2019-09-05 at 4 05 03 PM

**Comment by christiemolloy** [comment]
For the multiple typeahead select input when I have Alabama filtered and then I search for Alabama again and add it, both of them delete and it wipes Alabama clear? I don't know if this is the expected behavior?

Screen Shot 2019-09-05 at 4 06 57 PM

**Comment by christiemolloy** [comment]
When the menu expands up, should the arrow rotate @mcarrano 

Screen Shot 2019-09-05 at 4 09 32 PM

**Comment by kmcfaul** [comment]
@jschuler autocomplete is set to off for all selects, so I'm unsure why it would be showing up. Which browser are you using?

@christiemolloy `isCreatable` handles whether the user can add inputs, but will only do so temporarily without `onCreateOption` defined. `onCreateOption` is a callback for the user to store the new option in the overall options list, and is used together with isCreateable. 

Re: multi-typeahead, it looks like you are selecting Alabama twice, first which selects it and second which unselects it and removes it form the chip group.

**Comment by jschuler** [comment]
@kmcfaul strange, Chrome 76.0.3809.132. If no one else sees this then it might be something on my end

**Comment by kmcfaul** [comment]
That's my version of Chrome and I'm not seeing the autocomplete on the surge. Very odd. Anyone else seeing it?

**Comment by christiemolloy** [comment]
I see autocomplete in Chrome @kmcfaul @jschuler

**Comment by jschuler** [comment]
@kmcfaul do you have autofill enabled? `chrome://settings/addresses`

**Comment by kmcfaul** [comment]
I do have it enabled.

**Comment by jschuler** [comment]
Might have to wrap in form tags as this comment suggests https://gist.github.com/niksumeiko/360164708c3b326bd1c8#gistcomment-3004386

**Comment by kmcfaul** [comment]
That seems less than ideal, html structure wise. Will try it out though, hopefully shouldn't change anything visually.

**Comment by jschuler** [comment]
@kmcfaul ok, i don't recall seeing autocomplete before, i wonder why it's showing now

**Comment by mcarrano** [comment]
Regarding your question @christiemolloy:

> When the menu expands up, should the arrow rotate @mcarrano

I don't feel strongly about this, but I think we should just be consistent with what we do for other dropdown or select instances.

**Comment by kmcfaul** [comment]
I think it makes sense to have the arrow icon change to an upward direction on direction toggle. This is more of a quick bugfix than related to this particular PR though. I can put up a different PR for it.

**Comment by jschuler** [comment]
@kmcfaul wrapping inputs with form elements fixed the auto complete issue for me

**Pull request merged**